src/Raven.Studio/typescript/models/database/tasks/ongoingTaskQueueEtlEditModel.ts

```diff
@@ -0,0 +1,208 @@
+﻿/// <reference path="../../../../typings/tsd.d.ts"/>
+import ongoingTaskEditModel = require("models/database/tasks/ongoingTaskEditModel");
+import ongoingTaskQueueEtlTransformationModel = require("models/database/tasks/ongoingTaskQueueEtlTransformationModel");
+import jsonUtil = require("common/jsonUtil");
+import QueueBroker = Raven.Client.Documents.Operations.ETL.Queue.QueueBroker;
+
+class queueOptionsModel {
+    queueName = ko.observable<string>();
+    deleteProcessedDocuments = ko.observable<boolean>();
+
+    validationGroup: KnockoutObservable<any>;
+    dirtyFlag: () => DirtyFlag;
+
+    constructor(name: string, deleteSource: boolean) {
+        this.queueName(name);
+        this.deleteProcessedDocuments(deleteSource);
+
+        this.initValidation();
+
+        this.dirtyFlag = new ko.DirtyFlag([
+            this.queueName,
+            this.deleteProcessedDocuments,
+        ], false, jsonUtil.newLineNormalizingHashFunction);
+    }
+
+    private initValidation() {
+        this.queueName.extend({
+            required: true
+        });
+
+        this.validationGroup = ko.validatedObservable({
+            queueName: this.queueName
+        });
+    }
+
+    static empty() {
+        return new queueOptionsModel("", false);
+    }
+}
+
+abstract class ongoingTaskQueueEtlEditModel extends ongoingTaskEditModel {
+    
+    connectionStringName = ko.observable<string>();
+        
+    allowEtlOnNonEncryptedChannel = ko.observable<boolean>(false);
+    transformationScripts = ko.observableArray<ongoingTaskQueueEtlTransformationModel>([]);
+
+    showEditTransformationArea: KnockoutComputed<boolean>;
+
+    transformationScriptSelectedForEdit = ko.observable<ongoingTaskQueueEtlTransformationModel>();
+    editedTransformationScriptSandbox = ko.observable<ongoingTaskQueueEtlTransformationModel>();
+    
+    optionsPerQueue = ko.observableArray<queueOptionsModel>();
+    
+    validationGroup: KnockoutValidationGroup;
+    dirtyFlag: () => DirtyFlag;
+
+    get studioTaskType(): StudioTaskType {
+        return "KafkaQueueEtl";
```

Comment:
is it correct?

Reply:
I think you should remove this method - it is misleading.

Reply:
removed